### ISSUE TITLE
Submit page hits

### ISSUE THREAD

**Issue by vivoconunxino** [mautic/api-library#139]
I'd like to be able to mimic the page visit through the API. As far as I've seen it isn't possible yet. It would be great some king of call like 

$data['url'] = '/something';
$data['lead_id'] = 713;
$data['ip_id'] = 1;

// Create the page hit 
$response = $pageHit->create($data);

**Comment by vivoconunxino** [comment]
I could try it myself if you gave me some hint about where to start

**Comment by vivoconunxino** [comment]
Mmmm sorry escopez, but I can't really find out which url I shoud point out to tell mautic to increase a hit to a page (which doesnt really exists) '/something'.
I've tried a call to 

> http://localhost/mautic/index.php/mtracking.gif?page_url=something&email=sample@gmail.com

which doesn't work

**Comment by vivoconunxino** [comment]
Steeps:
In my localhost/mautic/index.php
- I've created a new Page 'something'
- From custom fields, I've changed the email field to unique and publicly available
- from another browser, went to http://localhost/mautic/index.php/mtracking.gif?page_url=something&email=sample@gmail.com
- from phpmyadmin, page_hits has the same number of hits than before

**Comment by escopecz** [comment]
that's odd. It should work. Try to debug around `PageModel::pageHit()`

**Comment by vivoconunxino** [comment]
Also, Logs doesn't register anything about.
Hi guess you mean PageModel::hitPage(). I'm going to take a look.

**Comment by vivoconunxino** [comment]
@escopecz is returning null at the start of the function:

> // Don't skew results with user hits
>         if (!$this->security->isAnonymous()) {
>             $this->logger->addError('---isAnonymous');
> 
>             return;
>         }

**Comment by escopecz** [comment]
See https://mautic.org/docs/en/pages/troubleshooting.html

**Comment by vivoconunxino** [comment]
yep, same browser, my fault, still does not work from another one but I couldn't tell you the point where it is failing

**Comment by vivoconunxino** [comment]
well... here must be the problem:
```
// Get lead if required
        if (null == $lead) {
            $lead = $this->leadModel->getContactFromRequest($query, $this->trackByFingerprint);
        }

        if (!$lead || !$lead->getId()) {
            // Lead came from a non-trackable IP so ignore
            return;
        }

```
getContactFromRequest is returning null
